src/routing_node.rs

```diff
@@ -836,6 +840,86 @@ impl RoutingNode {
                                   refresh_our_own_group: bool) -> RoutingResult {
         unimplemented!()
     }
+
+    fn handle_cache_get(&mut self, request : ExternalRequest,
+                        from_authority : Authority, to_authority : Authority,
+                        response_token : SignedToken) -> RoutingResult {
+        let name = match request.clone() {
+            ExternalRequest::Get(data_request) => {
+                match data_request {
+                    DataRequest::StructuredData(_, _) => return Err(From::from(ResponseError::NoData)),
+                    DataRequest::ImmutableData(data_name, _) => { data_name },
+                    DataRequest::PlainData(data_name) => { data_name }
+                }
+            },
+            _ => return Err(RoutingError::Response(ResponseError::InvalidRequest))
+        };
+
+        let response_data: Data;
+        match self.data_cache.get(&name) {
+            Some(data) => {
+                match request {
+                    ExternalRequest::Get(data_request) => {
+                        match data_request {
+                            DataRequest::StructuredData(_, _) =>
+                                return Err(From::from(ResponseError::NoData)),
+                            DataRequest::ImmutableData(data_name, data_type) => {
+                                response_data = Data::ImmutableData(
+                                    ImmutableData::new(data_type,
+                                                       data.iter().map(|&x|x).collect::<Vec<_>>()))
+                            },
+                            DataRequest::PlainData(data_name) => {
+                                response_data = Data::PlainData(
+                                    PlainData::new(data_name,
+                                                   data.iter().map(|&x|x).collect::<Vec<_>>()))
+                            }
+                        }
+                    },
+                    _ => return Err(RoutingError::Response(ResponseError::InvalidRequest))
+             };
+          },
+          None => return Err(From::from(ResponseError::NoData))
+        };
+        let routing_message = RoutingMessage {
+            from_authority : Authority::ManagedNode(self.core.id().name()),
+            to_authority   : Authority::ManagedNode(from_authority.get_location().clone()),
+            content        : Content::ExternalResponse(
+                ExternalResponse::Get(response_data, response_token)
+            ),
+        };
+
+        match SignedMessage::new(Address::Node(self.core.id().name()),
+            routing_message, self.core.id().signing_private_key()) {
+                Ok(signed_message) => ignore(self.send(signed_message)),
+                Err(e) => return Err(RoutingError::Cbor(e)),
+            };
+        Ok(())
+    }
+
+    fn handle_cache_put(&mut self, request : ExternalRequest,
```

Comment:
Here just try and insert to lru_time_cache. No need to return errors etc. it's a best effort.